pilot/pkg/xds/ads.go

```diff
@@ -870,6 +870,10 @@ func (s *DiscoveryServer) startPush(req *model.PushRequest) {
 			log.Debugf("Starting new push while %v were still pending", currentlyPending)
 		}
 	}
+
+	t := time.Since(req.DebounceStart)
+	debounceTime.Record(t.Seconds())
```

Comment:
Is it feasible to just handle this entirely in the debounce() function? it seems leaky to put this in PushRequest

Reply:
updated